src/grt/src/fastroute/include/FastRoute.h

```diff
@@ -590,6 +591,9 @@ class FastRouteCore
       vertical_blocked_intervals_;
   std::unordered_map<Tile, interval_set<int>, boost::hash<Tile>>
       horizontal_blocked_intervals_;
+
+  std::set<std::pair<int, int>> h_used_ggrid_;
+  std::set<std::pair<int, int>> v_used_ggrid_;
```

Comment:
Perhaps using `std::unordered_set` instead of `std::set`? If I understand correctly, your code doesn't care about the ordering, and an `unordered_set` will have better runtime to insert the elements.

You can try it with a large public design and check if the runtime is better and if the result keeps the same.

Reply:
For the public designs (ng45/bp and ng45/ariane136) I didn't find much difference in the runtime when I use `std::set` and `std::unordered_set` (I think it's due to the small number of ggrids that are modified in each iteration), even so I am going to add that modification since in private designs there may be a greater impact.